### ISSUE TITLE
Timer was not added to the SceneTree / `add_child()` failed

### ISSUE THREAD

**Issue by thygrrr** [Maran23/script-ide#16]
When I run Script-IDE, I often see this error, both at startup as well as on script reloads or during runtime in general

This happens on 4.1.2.stable and 4.2.beta
```
Godot Engine v4.1.2.stable.official (c) 2007-present Juan Linietsky, Ariel Manzur & Godot Contributors.
--- Debug adapter server started ---
--- GDScript language server started ---
  Timer was not added to the SceneTree. Either add it or set autostart to true.
  Parent node is busy setting up children, `add_child()` failed. Consider using `add_child.call_deferred(child)` instead.
  scene/main/window.cpp:1565 - Condition "!is_inside_tree()" is true.
```

I was able to work around the "add_child" issue by calling deferred `_setup` and `_teardown` functions fron `_enter_tree` and `_exit_tree` (deferred add child calls themselves are problematic because the setup code in `_enter_tree` refers to some of the added nodes directly)

The timer issue remains, nothing in Script-IDE's code seems to directly reference timers, but I think it may be something it does internally. There is no such error message when I remove the plugin.

**Comment by hydescarf** [comment]
Just to chip in for the "Timer" issue but here's a possible fix but on a different add-on: https://github.com/MakovWait/godot-script-tabs/issues/3
TL;DR: Emitting `item_selected` signal on `ItemList` is probably causing the error, `call_deferred` it should make the error disappear.

Actually got the `call_deferred` hint from your pull request :D

**Comment by Maran23** [comment]
Fixed by https://github.com/Maran23/script-ide/pull/19
